core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -471,6 +473,8 @@ public boolean apply(IndexWarmersMetaData.Entry warmer) {
     /**
      * Returns indexing routing for the given index.
      */
+    // TODO: This can be moved to IndexNameExpressionResolver too, but this means that we will support wildcards and other expressions
+    // in the index,bulk,update and delete apis.
```

Comment:
I tried to move it, but it is relative large change on top of this PR. Can this be done in a different PR?